src/components/Global/WalletHeader/index.tsx

```diff
@@ -0,0 +1,173 @@
+'use client'
+
+import { Wallet_ICON } from '@/assets'
+import PeanutWalletIcon from '@/assets/icons/small-peanut.png'
+import { Button, Card } from '@/components/0_Bruddle'
+import { useAuth } from '@/context/authContext'
+import { useWallet } from '@/context/walletContext'
+import { IWallet, WalletProviderType } from '@/interfaces'
+import { printableUsdc, shortenAddressLong } from '@/utils'
+import { useWeb3Modal } from '@web3modal/wagmi/react'
+import Image from 'next/image'
+import { useMemo, useState } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { useAccount } from 'wagmi'
+import CopyToClipboard from '../CopyToClipboard'
+import Icon from '../Icon'
+import Modal from '../Modal'
+
+interface WalletEntryCardProps {
+    wallet: IWallet
+    isActive?: boolean
+    onClick?: () => void
+}
+
+const WalletHeader = () => {
+    const [showModal, setShowModal] = useState(false)
+    const { wallets, selectedWallet, setSelectedWallet, isConnected } = useWallet()
+    const { open: openWeb3Modal } = useWeb3Modal()
+    const { connector } = useAccount()
+
+    // sort wallets to add active wallet at the top
+    const sortedWallets = useMemo(() => {
+        return [...wallets].sort((a, b) => {
+            if (a.address === selectedWallet?.address) return -1
+            if (b.address === selectedWallet?.address) return 1
+            return 0
+        })
+    }, [wallets, selectedWallet])
```

Comment:
Issue, blocking: This wallet ordering seems to be wrong - We should show the wallets in the home screen by this order:

left-most: Peanut wallet
2nd: 2nd added external wallet
3rd: 3rd added external wallet 

etc..

The active wallet should be the one selected, but not the left-most one